src/components/Search/CompareButton.tsx

```diff
@@ -1,20 +1,48 @@
 import Button from '@mui/material/Button';
 
+import { useAppSelector } from '../../hooks/app';
+import { SearchStyles } from '../../styles';
+import CancelButton from './CancelButton';
+
 interface CompareButtonProps {
   label: string;
+  hasNonEditableState: boolean;
+  onCancel: () => void;
+  onSetDisplayForm: (display: boolean) => void;
+  formIsDisplayed: boolean;
```

Comment:
Try to think about the "point of view" of this component.

All it needs to do its work is:
* should the cancel button be displayed => prop `hasCancelButton: boolean`
* was the cancel button clicked => onCancel

Therefore you can remove `hasNonEditableState` and `formIsDisplayed` (its caller can use them to pass `hasCancelButton` of course), and also `onSetDisplayForm` because the caller will handle this in its `onCancel` callback.

Reply:
Right, right. Rushed through this too quickly Thanks!